Integration branch for visual studio fixes and SAML login

jyotibhalerao opened the pull request:
Code changes for below points:
1)Authentication fix that impacted hosted customers.
2)Additional log messages.
3)Fix for PLUG-513 - Results browsing and not being able to see CxPath, CxGraph.
4)Not able to install VS2022 plugin on VS 2022 version lower than 17.0.1.
5)The flag that enabled lib64 zip.
6)Support for complex solution (For ex. Open Web site code directly without a solution or scanning a folder of a solution).
7)Unbind so that if user open another solution closing the current one, bind the operation will work successfully.
8)Saml login.

jyotibhalerao committed:
Update BindProjectLogic.cs

Code Changes for plug-488 subtask Clear bound project when switching to another solution.

jyotibhalerao committed:
Code changes for subtask PLUG-497 complex solution support

Code changes for subtask PLUG-497 complex solution support like websites in VS2019 and VS2022.

Commit:
add dotGitignore, Microsoft.VisualStudio.Shell.14.0 in packages.config

Commit:
Build 2019 and 2022 using msbuild tool and its project file.

jyotibhalerao committed:
Integration branch for visual studio fixes and tasks

Code changes for below points:
1)Authentication fix that impacted hosted customers.
2)Additional log messages.
3)Fix for PLUG-513 - Results browsing and not being able to see CxPath, CxGraph.
4)Not able to install VS2022 plugin on VS 2022 version lower than 17.0.1.
5)The flag that enabled lib64 zip.
6)Support for complex solution (For ex. Open Web site code directly without a solution or scanning a folder of a solution).
7)Unbind so that if user open another solution closing the current one, bind the operation will work successfully.
8)Changed plugin icon and more info redirecting to visual studio plugin site.

jyotibhalerao committed:
Changed build version to 9 .00.14

Changed build version to 9 .00.14 in manifest file

jyotibhalerao committed:
Added plugin icon in CxActionShared->Resources

Added new plugin icon in CxActionShared->Resources

jyotibhalerao committed:
Change for plugin icon and shared resources

Changes for plugin icon and shared resources

SubhadraSahoo reviewed (changes requested):
I have reviewed once. But you can ask Umesh also review once . Some places I see added code itself commented. Remove them.

apoorvasingh5 committed:
SAML login issue.

apoorvasingh5 committed:
Version Increment

apoorvasingh5 committed:
Update BrowserForm.cs

apoorvasingh5 committed:
BrowserForm.cs update to hide authorization code from logs.

jyotibhalerao committed:
Fixed issue LogOut for authentication modes 'access_control_ie' and 'username_password' is crashing Visual studio instance for VS19

Fixed issue LogOut' for authentication modes 'access_control_ie' and 'username_password' is crashing visual studio instance for VS19  and VS22 and added log statement and showing message if no vulnerabilities found on scan completion .

jyotibhalerao committed:
Merge branch 'build-issues-intg' of https://github.com/checkmarx-ltd/VisualStudio into build-issues-intg

jyotibhalerao committed:
Changed build version from 9.00.15 to 9.00.16

Changed build version from 9.00.15 to 9.00.16 in manifest file.

jyotibhalerao committed:
Fixed bugs of build 9.00.16

Fixed below bugs:
1) Added log statement on scan completion.
2) Login page when clicked on close button, removed login failed message.
3) Removed showing message box for “There are no vulnerabilities found” on vulnerabilities found.
4) Transferred some repetitive log statements for extracting files and folders from info mode to debug mode.

jyotibhalerao committed:
Corrected condition for login page close button click and if scan data is zero.

Corrected condition for login page close button click and if scan data is zero.

apoorvasingh5 committed:
Addition of Post data of the initial authentication request in the logs at debug level.

jyotibhalerao committed:
Removed commented and extra log and added post data

Made changes for below points:
1.	Removed ‘known issues’ log message
2.	Removed comments log message
3.	Post data of the initial authentication request to be logged at debug level.

jyotibhalerao committed:
Uncommented code for message box There are no vulnerabilities to show.

Uncomment code for message box There are no vulnerabilities to show and  load login data from loadSaved().

jyotibhalerao committed:
Removed if condition in added for testing purpose

Removed if condition in added for testing purpose. Missed to remove it.

cx-umesh-waghode reviewed (approved):
Looks good. QA cycle completed.

SubhadraSahoo reviewed (approved):
Looks good
